### ISSUE TITLE
Deleting and restoring usergroup leads to loss of link between user/usergroup

### ISSUE THREAD

**Issue by openmmp** [Icinga/icingaweb2-module-director#2914]
## Expected Behavior
When I delete a usergroup and afterwards undo all my changes via activity log, the former status should be _fully_ restored, that is including the link between any users and the deleted usergroup.

## Current Behavior
When I delete a usergroup and afterwards undo this action via activity log, the former status is not _fully_ restored, since the link between user and usergroup is missing. On the surface of the UI this link is shown as part of the user object. When deleting a usergroup, the deletion itself shows up in the activity log, yet the change in the user objects that depend on that group (which is the removal of the deleted group) is not. This may be due to the fact that these links are stored in a separate database table.

## Possible Solution
When deleting a usergroup, the resulting removal of links to user objects are logged as changes to those user objects to the activity log and can thus be restored.

## Steps to Reproduce (for bugs)
- add a user template
- add a usergroup
- add a user connected to this group
- delete usergroup
- view user & activity log

## Your Environment

* Director version: 1.11.1
* Icinga Web 2 version: 2.12.1
* PHP 8.2.13

**Comment by raviks789** [comment]
ref/IP/60399

**Comment by raviks789** [comment]
@openmmp, this is an expected behavior. The link between the user and the user group is established on the user side. And once the group is deleted, the link is automatically deleted. 

While restoring the group, restoring the link might lead to conflict in memberships, as there might be users who do not belong to the group anymore. So restoring the link would create more issues. So instead, the solution would be to support apply rules for users and to reestablish this connection through the apply rule.

Also, asking for a confirmation while deleting the group would prevent deleting the user groups by mistake.

Please check the PR #2991, with the described solution.

**Comment by ngoeddel-openi** [comment]
Hi @raviks789, I am a colleague of @openmmp and could need some help understanding the Pull Request. For me it looks like there were basically just a few safety questions added before one can delete a `UserGroup`.
Or does this PR also solve the issue with restoring a UserGroup including its members? If yes, then I would like to understand it a bit better if you don't mind.

**Comment by raviks789** [comment]
@ngoeddel-openi no, user group members will not be restored once the group is deleted. Therefore, safety prompts are now included to ensure that the group is only deleted when you are absolutely certain it is no longer needed.

As already explained we do not want to restore the membership due to the following reason:
> While restoring the group, restoring the link might lead to conflict in memberships, as there might be users who do not belong to the group anymore. So restoring the link would create more issues. So instead, the solution would be to support apply rules for users and to reestablish this connection through the apply rule.

**Comment by openmmp** [comment]
By now, the provided PR including apply rules for user groups should work for us. Thank you, for that proposal!

Is it possible to include it in the coming release? Is there a release date already?
